blazar/enforcement/filters/max_reservation_length_filter.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) 2013 Bull.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import datetime, timedelta
+
+from blazar import exceptions
+from blazar.i18n import _
+from blazar.enforcement.filters import base_filter
+
+from oslo_config import cfg
+from oslo_log import log as logging
+
+
+DEFAULT_MAX_RESERVATION_LENGTH = 0
+DEFAULT_RESERVATION_EXTENTSION_WINDOW = -1
+
+
+LOG = logging.getLogger(__name__)
+
+
+class MaxReservationLengthException(exceptions.BlazarException):
+    code = 400
+    msg_fmt = _('Lease length of %(lease_length)s seconds be less than or '
+                'equal the maximum lease length of %(max_length)s seconds.')
+
+
+class MaxReservationUpdateWindowException(exceptions.BlazarException):
```

Comment:
Ah nice, it didn't occur to me that this functionality is tied to this concept of max reservation length! A much better place for it.